### PR TITLE
Support fp8 quantization

### PR DESCRIPTION
**Pull request by australopitek**
This commit enables float8_e5m2 and float8_e4m3fn dtypes in fx quantization and PT2E.

Motivation for using fp8 quantization instead of int8:
- it works better to run inference with the same datatype the model was trained with,
- fp8 can handle outliers better, which is one of the problems in LLMs activations.

The numerical recipe we want to use it for is fp8 inference:
- bgemms/gemms running in float8_e4m3fn,
- Per-Tensor-Quantization/Scaling,
- amax observer for measurement with input_backoff and weight_backoff.

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

File: test/quantization/pt2e/test_quantize_pt2e.py [hunk 1]
```diff
@@ -1131,14 +1133,16 @@ def validate(self, model: torch.fx.GraphModule) -> None:
...
+    def test_quantization_dtype(self, dtype):
+        class DtypeActQuantizer(Quantizer):
             def annotate(self, model: torch.fx.GraphModule) -> torch.fx.GraphModule:
-                # using int32 to simulate int16
-                int16_qspec = QuantizationSpec(
-                    dtype=torch.int16,
-                    quant_min=-2**15,
-                    quant_max=2**15 - 1,
+                info_fun = torch.iinfo if dtype == torch.int16 else torch.finfo
+                # using int32 to simulate dtype
```
Comment (jerryzh168) [inline test/quantization/pt2e/test_quantize_pt2e.py:1141]: nit: you can remove this comment now

**Review by jerryzh168 — APPROVED** [review]
we haven't decided the long term plan of how to support static quant for LLMs, but this PR is small and it can enable people to try out float8 so I think it's fine to land

File: test/quantization/pt2e/test_quantize_pt2e.py (hunk 1 above)
Comment (australopitek) [reply test/quantization/pt2e/test_quantize_pt2e.py]: Done

**Comment by australopitek** [comment]
@jerryzh168 , thank you for review and approval. I have fixed merge conflicts and remove unnecessary comment, but I don't see any CI run.
Do I need to trigger something manually, or someone from Meta has to do it?

**Pull request closed without merging**